src/wfs/src/main/java/org/geoserver/wfs/json/ComplexGeoJsonWriter.java

```diff
@@ -360,6 +438,58 @@ private void encodeProperty(Property property) {
         }
     }
 
+    private <K, V> Map<K, V> mergeMaps(Map<K, V> mapA, Map<K, V> mapB) {
+        if (mapA == null) {
+            return mapB;
+        } else if (mapB == null) {
+            return mapA;
+        }
+
+        Map<K, V> merged = new HashMap<>(mapA);
+        merged.putAll(mapB);
+        return merged;
+    }
+
+    /**
+     * This code tries to determine if the current complex attribute is an example of GML
+     * property/type alternation. The GML gives us pretty much no firm indication to recognize them,
+     * there is no substitution group or inheritance, there are attribute groups sometimes found in
+     * these constructs, but not mandatory and not always present at the schema level.
+     *
+     * <p>This code works by recognizing the common alternation nomenclature, that is:
+     *
+     * <ul>
+     *   <li>The attribute type is called ${name}PropertyType
+     *   <li>It contains a single element inside, which is in turn another complex attribute itself
+     *   <li>The contained element type is called ${name}Type or the property is called ${name}
+     * </ul>
```

Comment:
Well this look like the only safe path we can use when looking only at the XML \ XSD. In the future we may also want to add some information in the property user data that says that it was feature chained, hence an high probability of being a GML property type.

Reply:
Works for me